stan/math/prim/fun/unit_vector_free.hpp

```diff
@@ -18,11 +18,10 @@ namespace math {
  * @param x unit vector of dimension K
  * @return Unit vector of dimension K considered "free"
  */
-template <typename T>
-Eigen::Matrix<T, Eigen::Dynamic, 1> unit_vector_free(
-    const Eigen::Matrix<T, Eigen::Dynamic, 1>& x) {
+template <typename EigVec, require_eigen_vector_t<EigVec>* = nullptr>
+decltype(auto) unit_vector_free(EigVec&& x) {
```

Comment:
```suggestion
EigVec unit_vector_free(EigVec&& x) {
```
No need for `decltype(auto)` when we know the return type.

Reply:
One caveat I wanna check, if we return by value here and assign this to a `const auto&` is that legal? Also, micro-optim so not a great argument, but `decltype(auto)` can save a construction (though it's a move construction so not really a big deal)

https://godbolt.org/z/YWEoEx

Reply:
Yes it is legal. Saving a construction can also make a seemingly innocent code wrong. You actually have an example in the linked godbolt:

```
	return decltype(auto) from rvalue to auto&&:
		Default Constructor
		Destruct
```
There was no copy or move constructor call - The result got destructed!